tools/smb_file_server.rb

```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env ruby
+
+require 'pathname'
+require 'ruby_smb'
+
+# we just need *a* default encoding to handle the strings from the NTLM messages
+Encoding.default_internal = 'UTF-8' if Encoding.default_internal.nil?
+
+options = RubySMB::Server::Cli.parse(defaults: { share_path: '.' }) do |options, parser|
```

Comment:
What are your thoughts on overriding the default user/pass here to something more msf-y instead of RubySMB/password, or printing out the default credentials so folk know what the creds are? 👀

Reply:
Changed the username to `metasploit`. As for the default values, when the user specifies `--help`, the default for each is at the end of their description:

```
Usage: smb_file_server.rb [options]

Start a read-only SMB file server.

Options:
        --share-name SHARE_NAME      The share name (default: home)
        --[no-]anonymous             Allow anonymous access (default: true)
        --[no-]guests                Allow guest accounts (default: false)
        --[no-]smbv1                 Enable or disable SMBv1 (default: Enabled)
        --[no-]smbv2                 Enable or disable SMBv2 (default: Enabled)
        --[no-]smbv3                 Enable or disable SMBv3 (default: Enabled)
        --username USERNAME          The account's username (default: metasploit)
        --password PASSWORD          The account's password (default: password)
        --share-path SHARE_PATH      The path to share (default: .)
```